kubernetes.go

```diff
@@ -0,0 +1,90 @@
+package veneur
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type KubernetesDiscoverer struct {
+	clientset *kubernetes.Clientset
+}
+
+func NewKubernetesDiscoverer() (*KubernetesDiscoverer, error) {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &KubernetesDiscoverer{clientset}, nil
+}
+
+func (kd *KubernetesDiscoverer) GetDestinationsForService(serviceName string) ([]string, error) {
```

Comment:
Do we need to measure the time in this function or is it measured outside?

Reply:
It's measured as part of `RefreshDestinations`

Reply:
Great!